backends/pep517.py

```diff
@@ -21,99 +21,308 @@
+        lines[0] = "#!python"
+    with open(path, "w") as f:
+        f.write("\n".join(lines))
+
+
+def _inject_wheel_extras(whl_path):
+    with _extracted_wheel_dir(whl_path) as extracted_wheel_dir:
+        _build_and_inject_ac_index(BUILD_DIR, extracted_wheel_dir)
+        _inject_scripts(extracted_wheel_dir)
+
+
+def _inject_editable_wheel_extras(whl_path):
+    with _extracted_wheel_dir(whl_path) as extracted_wheel_dir:
+        _inject_scripts(extracted_wheel_dir)
+
+
+@contextlib.contextmanager
+def _extracted_wheel_dir(whl_path):
+    _create_dir_if_not_exists(BUILD_DIR)
+    unpack_wheel_dir = os.path.join(BUILD_DIR, "unpacked_wheel")
     _remove_dir_if_exists(unpack_wheel_dir)
-    print('Adding auto-complete index into wheel')
     with _open_wheel(whl_path, unpack_wheel_dir) as extracted_wheel_dir:
-        os.rename(
-            ac_index_build_name,
-            os.path.join(extracted_wheel_dir, 'awscli', 'data', 'ac.index')
+        yield extracted_wheel_dir
+
+
+def _build_and_inject_ac_index(build_dir, extracted_wheel_dir):
+    ac_index_build_name = _build_ac_index(build_dir)
+    print("Adding auto-complete index into wheel")
+    os.rename(
+        ac_index_build_name,
+        os.path.join(
+            extracted_wheel_dir,
+            "awscli",
+            "data",
+            "ac.index",
+        ),
+    )
+
+
+def _build_ac_index(build_dir, rebuild=True):
+    from awscli.autocomplete.generator import generate_index
+
+    ac_index_build_name = os.path.join(build_dir, "ac.index")
+    if rebuild:
+        _remove_file_if_exists(ac_index_build_name)
+    elif os.path.exists(ac_index_build_name):
+        return ac_index_build_name
+
+    print("Generating auto-complete index")
+    generate_index(ac_index_build_name)
+    return ac_index_build_name
+
+
+def _inject_scripts(extracted_wheel_dir):
+    # There are two directories generated by flit:
+    # 1) awscli/
+    # 2) awscli-version.dist-info/
+    # The easiest way to generate a data directory is to
+    # find the dist-info directory and replace dist-info with
+    # data.
+    # There shold be exactly one, if not then something has gone horribly wrong
+    # and crashing here is appropriate.
+    dist_info_dir = [
+        d for d in os.listdir(extracted_wheel_dir) if "dist-info" in d
+    ][0]
+    data_dir = os.path.join(
+        extracted_wheel_dir,
+        dist_info_dir.replace("dist-info", "data"),
+    )
+    _create_dir_if_not_exists(data_dir)
+    scripts_dir = os.path.join(data_dir, "scripts")
+    _create_dir_if_not_exists(scripts_dir)
+    for script in SCRIPTS:
+        target = os.path.join(scripts_dir, script)
+        print(f"Injecting script {script} -> {target}")
+        shutil.copy2(
+            os.path.join(BIN_DIR, script),
+            target,
         )
+        _rewrite_shebang(target)
+    _create_record_file(extracted_wheel_dir, dist_info_dir)
+
+
+def _inject_record_entry(target, base_dir, dist_info_dir):
```

Comment:
I'm not sure if this function is used anymore. It looks like the records file is recreated once now